AK/BigInts.h

```diff
@@ -0,0 +1,487 @@
+/*
+ * Copyright (c) 2021, Leon Albrecht <leon2002.la@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Checked.h>
+#include <AK/Concepts.h>
+#include <AK/Format.h>
+#include <AK/NumericLimits.h>
+#include <AK/StdLibExtraDetails.h>
+#include <AK/StdLibExtras.h>
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
+
+namespace AK {
+
+template<typename T>
+class UBigInt;
+template<typename T>
+constexpr bool Detail::IsIntegral<UBigInt<T>> = true;
+template<typename T>
+constexpr bool Detail::IsUnsigned<UBigInt<T>> = true;
+template<typename T>
+constexpr bool Detail::IsSigned<UBigInt<T>> = false;
+
+template<typename T>
+class UBigInt {
+    // this sadly cannot be `require(...)`-d
+    static_assert(sizeof(T) >= sizeof(u64), "Creating BigInts of types smaller than u64 is useless");
```

Comment:
why not?
I'd expect this to be fine:
```suggestion
template<typename T>
requires(sizeof(T) >= sizeof(u64))
class UBigInt {
```

(you'll have to put it on the forward declaration above too)

Reply:
this causes a error, due to this class being used recursively and therefore "incomplete" in the requirement of the forward declaration, which is a thing GCC doesn't like